psydac/linalg/block.py

```diff
@@ -500,12 +500,12 @@ class BlockMatrix( BlockLinearOperator, Matrix ):
     #--------------------------------------
     # Abstract interface
     #--------------------------------------
-    def toarray( self, *, order='C', with_pads=False ):
+    def toarray( self, *, order='C' ):
         """ Convert to Numpy 2D array. """
         return self.tosparse(order=order).toarray()
 
     # ...
-    def tosparse( self, *, order='C', with_pads=False ):
+    def tosparse( self, *, order='C' ):
```

Comment:
It's interesting that Codacy does not complain about the fact that these methods have a signature different from that of the base class, which is
```
def toarray(self, **kwargs):
```
So did Codacy complain if you kept the same signature in the classes `StencilVector` and `StencilMatrix`?

Reply:
it did not complain

Reply:
If it did not complain, why did you not keep this signature in those classes?
```python
def toarray(self, *, with_pads=False):
```
This is more informative than `**kwargs`, which is only needed in the base class.

Reply:
for some reason, it does not complain about the Block Matrix, but it does for the StencilMatrix